core/trino-main/src/main/java/io/trino/sql/rewrite/ShowQueriesRewrite.java

```diff
@@ -389,13 +389,20 @@ protected Node visitShowColumns(ShowColumns showColumns, Void context)
             if (!metadata.schemaExists(session, new CatalogSchemaName(tableName.getCatalogName(), tableName.getSchemaName()))) {
                 throw semanticException(SCHEMA_NOT_FOUND, showColumns, "Schema '%s' does not exist", tableName.getSchemaName());
             }
-            Optional<ConnectorViewDefinition> view = metadata.getView(session, tableName);
-            Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
-            if (view.isEmpty() && tableHandle.isEmpty()) {
+
+            boolean isMaterializedView = metadata.getMaterializedView(session, tableName).isPresent();
+            boolean isView = false;
+            Optional<TableHandle> tableHandle = Optional.empty();
+            if (!isMaterializedView) {
+                isView = metadata.getView(session, tableName).isPresent();
+                tableHandle = metadata.getTableHandle(session, tableName);
```

Comment:
i see you're saving `metadata.getView` when `isMaterializedView`
why aren't you saving `metadata.getTableHandle` call when something `isView`?